rados/ioctx.go

```diff
@@ -231,6 +231,23 @@ func (ioctx *IOContext) GetPoolName() (name string, err error) {
 	}
 }
 
+// SetLocator sets the key for mapping objects to pgs within an io context.
+// Until a different locator key is set, all objects in this io context will be placed in the same pg.
+// To reset the locator, an empty string must be set.
+//
+// Implements:
+//  void rados_ioctx_locator_set_key(rados_ioctx_t io, const char *key);
+func (ioctx *IOContext) SetLocator(locator string) {
+	var c_loc *C.char
+	if len(locator) > 0 || locator != "" {
```

Comment:
This check seems redundant to me.
Also, if passing a nil-for-null is required wouldn't it be clearer to do something more like:
```golang
var cloc *C.char
if locator != "" {
    cloc = C.CString(locator)
    defer C.free(unsafe.Pointer(c_loc))
}
C.rados_ioctx_locator_set_key(ioctx.ioctx, c_loc)
```

This brings up one more concern, which is what to do if empty string ("") has a different meaning in this api. IOW, If "" is valid as input to the ceph api versus us treating it as 'unset'. I don't know enough about this aspect of ceph myself at the moment to make an educated suggestion.

Reply:
1.) The first part is really redundant . I will fix that.
2.) As far as I understand we cant use "" for resetting the namespace, rather we must remove the pointer. 
> .... or NULL use the * default namespace [Link](https://github.com/ceph/ceph/blob/master/src/include/rados/librados.h#L1027)

Reply:
Can we test that? In many other apis I've checked so far it seems that "" and a null string are treated equivalently (even at the ceph api level). But I would hate for there to be a case where "" is valid and we force it to be treated as null string. If we have to, I'd also be open to having `ResetLocator()` (taking no args) in our api if it makes sense.